### ISSUE TITLE
Problems with arguments

### ISSUE THREAD

**Issue by kathy-snider** [nipreps/nibabies#77]
I am running version 0.1.0. I first ran with --help, to get the arguments to use, and got the following:
usage: nibabies [-h] [--version] [--skip_bids_validation]
                [--participant-label PARTICIPANT_LABEL [PARTICIPANT_LABEL ...]]
                [-t TASK_ID] [--echo-idx ECHO_IDX] [--bids-filter-file FILE]
                [--anat-derivatives PATH] [--bids-database-dir PATH]
                [--nprocs NPROCS] [--omp-nthreads OMP_NTHREADS]
                [--mem MEMORY_GB] [--low-mem] [--use-plugin FILE]
                [--anat-only] [--boilerplate_only] [--md-only-boilerplate]
                [--error-on-aroma-warnings] [-v]
                [--ignore {fieldmaps,slicetiming,sbref,t2w,flair} [{fieldmaps,slicetiming,sbref,t2w,flair} ...]]
                [--longitudinal]
                [--output-spaces [OUTPUT_SPACES [OUTPUT_SPACES ...]]]
                [--bold2t1w-init {register,header}] [--bold2t1w-dof {6,9,12}]
                [--force-bbr] [--force-no-bbr] [--medial-surface-nan]
                [--dummy-scans DUMMY_SCANS] [--random-seed _RANDOM_SEED]
                [--use-aroma]
                [--aroma-melodic-dimensionality AROMA_MELODIC_DIM]
                [--return-all-components]
                [--fd-spike-threshold REGRESSORS_FD_TH]
                [--dvars-spike-threshold REGRESSORS_DVARS_TH]
                [--skull-strip-template SKULL_STRIP_TEMPLATE]
                [--skull-strip-fixed-seed]
                [--skull-strip-t1w {auto,skip,force}] [--fmap-bspline]
                [--fmap-no-demean] [--use-syn-sdc] [--force-syn]
                [--fs-license-file FILE] [--fs-subjects-dir PATH]
                [--no-submm-recon] [--cifti-output [{91k,170k}] |
                --fs-no-reconall] [--output-layout {bids,legacy}]
                [-w WORK_DIR] [--clean-workdir] [--resource-monitor]
                [--reports-only] [--config-file FILE] [--write-graph]
                [--stop-on-first-crash] [--notrack]
                [--debug {compcor,registration,fieldmaps,all} [{compcor,registration,fieldmaps,all} ...]]
                [--sloppy] [--age-months AGE_MONTHS]
                [--segmentation-atlases-dir SEGMENTATION_ATLASES_DIR]
                bids_dir output_dir {participant}

....followed by details of args, including:
  --participant-label PARTICIPANT_LABEL [PARTICIPANT_LABEL ...], --participant_label PARTICIPANT_LABEL [PARTICIPANT_LABEL ...]
                        a space delimited list of participant identifiers or a
                        single identifier (the sub- prefix can be removed)
                        (default: None)

So I ran the program, and got this error:
+ docker run --participant-label 06IND001B --age-months 12 --segmentation-atlases-dir /mnt/max/shared/sniderk/templates/08-11 mgxd/nibabies:0.1.0 /mnt/m ax/shared/sniderk/nibabies/niftis /mnt/max/shared/sniderk/nibabies/output participant
unknown flag: --participant-label
See 'docker run --help'.

As you can see, participant-label is listed in help. I do not know another way to specify the participant, since that is the only way shown.

**Comment by kathy-snider** [comment]
Note: I also ran with the positional args before the options, since the order is unclear: 
+ docker run mgxd/nibabies:0.1.0 /mnt/max/shared/sniderk/nibabies/niftis /mnt/max/shared/sniderk/nibabies/output participant --participant-label 06IND001B --age-months 12 --segmentation-atlases-dir /mnt/max/shared/sniderk/templates/08-11

In that case, I got a usage dump followed by:
nibabies: error: Path does not exist: </mnt/max/shared/sniderk/nibabies/niftis>

However, I can do this:
sniderk@rushmore:nibabies$ ll /mnt/max/shared/sniderk/nibabies/niftis
total 8.0K
-rw-rw---- 1 sniderk fnl   43 Jun 21 09:29 dataset_description.json
drwxrwx--- 5 sniderk fnl 4.0K Jun  8 14:12 sub-06IND001B

**Comment by mgxd** [comment]
You will have to put the arguments after the `docker run <docker-options> mgxd/nibabies:0.1.0` command.

To share files between your local filesystem and a Docker container, you will have to add bind mounts. So in your case the command will look like:

```
docker run --rm -it \
-v /mnt/max/shared/sniderk:/rootbind \
mgxd/nibabies:0.1.0 /rootbind/nibabies/niftis /rootbind/nibabies/output participant \
--participant-label 06IND001B --age-months 12 --segmentation-atlases-dir /rootbind/templates/08-11
```

Where `/rootbind` in the container is an alias for `/mnt/max/shared/sniderk`.

However, just like fMRIPrep, you will still need to include a FreeSurfer license file through `--fs-license-file`.

**Comment by mgxd** [comment]
FWIW, I am currently working on a wrapper to handle the Docker/Singularity syntax for users.

**Comment by kathy-snider** [comment]
Nevermind - this is a user misunderstanding. The second order is correct - put the required args _before_ the options (in spite of the usage message), but yes, as you point out, I needed the -v mount!
+ docker run -v /mnt/max/shared/sniderk/nibabies/niftis:/bids_dir -v /mnt/max/shared/sniderk/nibabies/output:/output_dir mgxd/nibabies:0.1.0 /bids_dir /output_dir participant --participant-label 06IND001B --age-months 12 --segmentation-atlases-dir /mnt/max/shared/sniderk/templates/08-11
Traceback (most recent call last):
  File "/usr/local/miniconda/bin/nibabies", line 33, in <module>
    sys.exit(load_entry_point('nibabies', 'console_scripts', 'nibabies')())
  File "/src/nibabies/nibabies/cli/run.py", line 17, in main
    parse_args()
  File "/src/nibabies/nibabies/cli/parser.py", line 706, in parse_args
    from ..utils.bids import validate_input_dir
ImportError: cannot import name 'validate_input_dir' from 'nibabies.utils.bids' (/src/nibabies/nibabies/utils/bids.py)

Got the new error above, but still working through it.

**Comment by mgxd** [comment]
See https://github.com/nipreps/nibabies/issues/73#issuecomment-864191337

BTW, you will need to ensure your segmentation atlases are also mounted.

**Comment by kathy-snider** [comment]
Right. Mounted those, too. Now am having a problem with validation?

+ docker run -v /mnt/max/shared/sniderk/nibabies/niftis:/bids_dir -v /mnt/max/shared/sniderk/nibabies/output:/output_dir -v /mnt/max/shared/sniderk/freesurfer_license.txt:/opt/freesurfer/license.txt -v /mnt/max/shared/sniderk/templates/08-11/JLF_templates:/templates mgxd/nibabies:0.1.0 /bids_dir /output_dir participant --participant-label 06IND001B --age-months 12 --segmentation-atlases-dir /templates
Traceback (most recent call last):
  File "/usr/local/miniconda/bin/nibabies", line 33, in <module>
    sys.exit(load_entry_point('nibabies', 'console_scripts', 'nibabies')())
  File "/src/nibabies/nibabies/cli/run.py", line 17, in main
    parse_args()
  File "/src/nibabies/nibabies/cli/parser.py", line 706, in parse_args
    from ..utils.bids import validate_input_dir
ImportError: cannot import name 'validate_input_dir' from 'nibabies.utils.bids' (/src/nibabies/nibabies/utils/bids.py)

Do I just need to clone the repo and mount something else? Or is there something missing from the docker image?

**Comment by mgxd** [comment]
The container is missing something, you can just add `--skip-bids-validation` to avoid the error.

**Comment by mgxd** [comment]
This should be fixed with the newest release (`0.1.1`)
